### ISSUE TITLE
`maybe` on root map does not work as expected

### ISSUE THREAD

**Issue by JoelSanchez** [metosin/spec-tools#100]
Hi, it seems using `maybe` on the first value to be validated, when it is a map, returns inconsistent results:

```
;; maybe map, nil value
;; expectation: true
;; result: false
(spec/valid?
 (data-spec/spec ::example
                 (maybe {:n number?}))
 nil)
=> false

:: maybe map, valid value
;; expectation: true
;; result: false
(spec/valid?
 (data-spec/spec ::example
                 (maybe {:n number?}))
 {:n 5})
=> false

;; map, valid value
;; expectation: true
;; result: true
(spec/valid?
 (data-spec/spec ::example
                 {:n number?})
 {:n 5})
=> true

;; nilable map, nil value
;; expectation: true
;; result: true
(spec/def ::n number?)
=> :repl/n
(spec/valid?
 (spec/nilable
  (spec/keys :req-un [::n]))
 {:n 5})
=> true
```

Am I missing something?

**Comment by ikitommi** [comment]
Good catch, the top-level `ds/maybe` was not supported. Should be fixes in #102 and deployed in `[metosin/spec-tools "0.6.0-20180130.060553-4"]`. Could you tests with that?

**Comment by JoelSanchez** [comment]
Thank you! Fixed
